### PR TITLE
Adds possibility to define portRef for extra ingresses

### PR DESCRIPTION
**Pull request by hfjn**
We have the use case were want multiple ingresses with different ingress Classes point to the same Container Port. Therefore, the underlying service needs to expose the same container port on different service ports. With the current setup that's not possible. Therefore, I added a possibility to define the `portRef` to allow this mapping.

### CODE REVIEW

File: charts/generic-service/ci/ingress-extra-values.yaml [hunk 1]
```diff
@@ -23,3 +23,10 @@ ingress:
         nginx.ingress.kubernetes.io/configuration-snippet: other-dummy-config;
       protocol: grpc
       timeoutSeconds: 5
+    extra-port:
+      portRef: other-port
+      domains: ['custom-port.example.com']
+      annotations:
+        nginx.ingress.kubernetes.io/server-snippet: other-dummy-config;
```
Comment (bastianeicher) [inline charts/generic-service/ci/ingress-extra-values.yaml]:
```suggestion
        nginx.ingress.kubernetes.io/configuration-snippet: other-dummy-config;
```

File: charts/generic-service/templates/ingress-extra.yaml [hunk 2]
```diff
@@ -61,12 +61,11 @@ spec:
               service:
                 name: {{ include "generic-service.fullname" $ }}
                 port:
-                {{- if $extra.port }}
-                  name: {{ include "generic-service.normalize-protocol" $extra.protocol }}-{{ $name }}
+                {{- if $extra.portRef }}
+                  name: {{ $extra.protocol }}-{{ $extra.portRef }}
+                {{- else if $extra.port }}
+                  name: {{ $extra.protocol }}-{{ $name }}
```
Comment (bastianeicher) [inline charts/generic-service/templates/ingress-extra.yaml]:
```suggestion
                {{- if $extra.portRef }}
                  name: {{ include "generic-service.normalize-protocol" $extra.protocol }}-{{ $extra.portRef }}
                {{- else if $extra.port }}
                  name: {{ include "generic-service.normalize-protocol" $extra.protocol }}-{{ $name }}
```

File: charts/generic-service/README.md [hunk 3]
```diff
@@ -122,7 +122,8 @@ app:
 | `ingress.istio.httpHeaders`                     | `{}`                        | Custom HTTP response headers                                                                             |
 | `ingress.istio.retries`                         | `{}`                        | [Istio retry policy](https://istio.io/docs/reference/config/networking/virtual-service/#HTTPRetry)       |
 | `ingress.extra.*.class`                         | same as `ingress.class`     | Additional ingress controller to use (not applicable if `ingress.istio.enabled`)                         |
-| `ingress.extra.*.port`                          | same as `ingress.port`      | Additional container port ingress traffic is routed to (not applicable if `ingress.istio.enabled`)       |
+| `ingress.extra.*.port`                          | same as `ingress.port`      | Additional service port ingress traffic is routed to (not applicable if `ingress.istio.enabled`)       |
```
Comment (bastianeicher) [inline charts/generic-service/README.md]:
```suggestion
| `ingress.extra.*.port`                          | same as `ingress.port`      | Additional container port ingress traffic is routed to (not applicable if `ingress.istio.enabled`)       |
```

File: charts/generic-service/values.schema.json [hunk 4]
```diff
@@ -650,7 +650,11 @@
               },
               "port": {
                 "type": "integer",
-                "description": "The container port ingress traffic is routed to; defaults to value of ingress.port if not set"
+                "description": "The service port ingress traffic is routed to; defaults to value of ingress.port if not set"
```
Comment (bastianeicher) [inline charts/generic-service/values.schema.json]:
```suggestion
                "description": "The container port ingress traffic is routed to; defaults to value of ingress.port if not set"
```

File: charts/generic-service/values.schema.json [hunk 5]
```diff
@@ -650,7 +650,11 @@
...
+              },
+              "portRef": {
+                "type": "string",
+                "description": "Reference to an existing extra ingress to replicate its port configuration."
```
Comment (bastianeicher) [inline charts/generic-service/values.schema.json]:
```suggestion
                "description": "Reference to the name of another extra ingress to share its container port"
```

File: charts/generic-service/README.md [hunk 6]
```diff
@@ -122,7 +122,8 @@ app:
...
+| `ingress.extra.*.portRef`                       |                             | Reference to existing extra ingress to replicate its port configuration.
```
Comment (bastianeicher) [inline charts/generic-service/README.md]:
```suggestion
| `ingress.extra.*.portRef`                       |                             | Reference to the name of another extra ingress to share its container port                               |
```

File: charts/generic-service/templates/ingress-extra.yaml [hunk 7]
```diff
@@ -61,12 +61,11 @@ spec:
...
                 port:
-                {{- if $extra.port }}
+                {{- if $extra.portRef }}
+                  name: {{ include "generic-service.normalize-protocol" $extra.protocol }}-{{ $extra.portRef }}
+                {{- else if $extra.port }}
                   name: {{ include "generic-service.normalize-protocol" $extra.protocol }}-{{ $name }}
                 {{- else }}
-                  {{- if $extra.protocol }}
-                    {{ fail "ingress.extra.*.protocol may not be set if ingress.extra.*.port is not set" }}
-                  {{- end }}
```
Comment (bastianeicher) [inline charts/generic-service/templates/ingress-extra.yaml]: Why remove this?

Comment (hfjn) [reply charts/generic-service/templates/ingress-extra.yaml]: woops

Comment (hfjn) [reply charts/generic-service/templates/ingress-extra.yaml]: Fixed

**Pull request merged**